pkg/v1alpha2/registry/access/remote_access_type_v1alpha2.go

```diff
@@ -0,0 +1,79 @@
+package access
+
+import (
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	log "github.com/sirupsen/logrus"
+	types2 "github.com/vmware/hamlet/api/types/v1alpha2"
+	"github.com/vmware/hamlet/pkg/v1alpha2/registry/resources"
+)
+
+// FederatedServiceObserver is an observer that receives notifications when
+// federated services are created, updated, or deleted.
+type FederatedServiceObserverV1Alpha2 interface {
+	// OnCreate is called when a new federated service is created.
+	OnCreate(providerId string, svc *types2.FederatedService) error
+
+	// OnUpdate is called when an existing federated service is updated.
+	OnUpdate(providerId string, svc *types2.FederatedService) error
```

Comment:
We have changed operations in the protobuf to UPSERT and DELETE. However, here we still have the three events (create, update, delete). Can we also update here to two events or how could we differentiate an update from a create if not?

Reply:
yes